doc/guide/fpga-update.md

```diff
@@ -0,0 +1,138 @@
+
+```
+LOG.TXT
+OPTIONS.TXT
+README.TXT
+```
+
+If the board has a RPI loader, visible because you'll see the file `INFO_UF2.txt` with the contents `Model: Raspberry Pi RP2`, you may need to reload the RP2040 as described in the [Sonata Board Updating Firmware]() section. This could
+be because the `RPI Boot` button was held down when plugging in the board.
+
+The Sonata board will print status and messages to the `LOG.TXT` which can be helpful for debugging. It should show the status of valid-looking bitstreams:
+
+```
+CRIT: FW_VER 0.1.2
+INFO: No bitstream in slot 0
+INFO: No bitstream in slot 1
+INFO: No bitstream in slot 2
+INFO: Using slot 2
+INFO: No bitstream in flash @ 1400000
+```
+
+### FPGA Config Led not coming on
+
+If the `FPGA Config` LED is not coming on, this could indicate the bitstream was designed for a different FPGA, or some other hardware issue. This should be troubleshooted with the OpenFPGALoader utility as described further down this page.
+
+### Ibex Boot LED not coming on
+
+If the `FPGA Config` LED is on but the `Ibex Boot` LED is not, it may be that you have programmed (or selected) a different bitstream than one that runs the CHERIoT demo. Try reloading the bitstream, and try power cycling the device.
+
+### Device Rebooting During/After Programming
+
+The Sonata board takes a fair amount of power (>500mA) from the USB interface, and should be connected via USB-C. Typically it is close enough to the USB 2.0 limits that it will work with the adapter most of the time, but if you are having reliability issues we recommend trying a different computer, ideally one with a USB-C port.
+
+### FPGA Programming via USB/JTAG
+
+If for some reason the mass storage programming isn't working, you can also use the built-in FTDI JTAG programming. This requires the setup described in [FPGA Programming](../dev/fpga-programming.md) to build openFPGALoader. Once built, you simply run:
+
+```sh
+openFPGALoader -c ft4232 sonata_top.bit
+```
+
+Note this requires the `udev` setup described in [FPGA Programming](../dev/fpga-programming.md). If you are lazy you can just run the command as root instead (not recommended, but can be helpful for troubleshooting on VMs):
+
+```sh
+sudo openFPGALoader -c ft4232 sonata_top.bit
+```
+You can also check if the flag `--read-register STAT` is available (newer than 0.12.2 is needed) which will print detailed information about the configuration status with a recent version of openFPGALoader. This is especially helpful if you are trying to understand why the `DONE` LED is not coming on:
+
+```
+CRC Error      No CRC error
+Part Secured   0
+MMCM lock      1
+DCI match      1
+EOS            0
+GTS CFG B      0
+GWE            0
+GHIGH B        0
+MODE           7
+INIT Complete  1
+INIT B         0
+Release Done   0
+Done           0
+ID Error       ID error
+DEC Error      0
+XADC Over temp 0
+STARTUP State  0
+Reserved       0
+BUS Width      x1
+Reserved       8
+```
```

Comment:
This is repeated from the FPGA programming section, maybe just leave it there for now?

Reply:
I made a short note you can use this feature for debug & linked to the fpga programming section.

Reply:
Ok, sounds good.